src/Nethermind/Nethermind.Runner/JsonRpc/CountingTextReader.cs

```diff
@@ -0,0 +1,94 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nethermind.Runner.JsonRpc
+{
+    public class CountingTextReader : TextReader
+    {
+        private readonly TextReader _innerReader;
```

Comment:
Why do we need the field? Maybe just inherit from TextReader and instantiate directly ?
Can we use this class in JsonRpcSocketClient instead of the STreamReader and utilize the counting there?

Reply:
We would have to inherit from StringReader or StreamReader, now using decorator pattern we can use it with any TextReader